src/xc_integrator/local_work_driver/device/cuda/kernels/zmat_vxc.cu

```diff
@@ -145,5 +145,87 @@ void zmat_gga_vxc( size_t            ntasks,
 
 
 
+
+
+
+
+
+template<int den_selector>
+__global__ void zmat_lda_vxc_kernel( size_t        ntasks,
+                                     XCDeviceTask* tasks_device ) {
+
+  const int batch_idx = blockIdx.z;
+  if( batch_idx >= ntasks ) return;
+
+  auto& task = tasks_device[ batch_idx ];
+  const auto npts            = task.npts;
+  const auto nbf             = task.bfn_screening.nbe;
+  const double* vrho_pos_device    = task.vrho_pos;
+  const double* vrho_neg_device    = task.vrho_neg;
+
+
+  const auto* basis_eval_device = task.bf;
+
+  auto* z_matrix_device = task.zmat;
+
+  const int tid_x = blockIdx.x * blockDim.x + threadIdx.x;
+  const int tid_y = blockIdx.y * blockDim.y + threadIdx.y;
+
+  if( tid_x < npts and tid_y < nbf ) {
+
+    const size_t ibfoff = tid_y * npts + tid_x;
+    const double factp = 0.5 * vrho_pos_device[tid_x];
+    const double factm = 0.5 * vrho_neg_device[tid_x];
+    switch ( den_selector ) {
```

Comment:
The only different between these two paths are a $\pm$, this can and should be simplified. e.g. 

```
constexpr auto fac_den = pos_density ? 1.0 : -1.0;
zmat[ioff] = a + fac_den * b;
```